### PR TITLE
chore(lint): cleanup unused variables in code

### PR DESCRIPTION
**Pull request by tech4him1**
Remove unused variables in code. I think this should get a fairly thorough review to make sure that there are not hidden bugs if we *should* have been using these variables.

### CODE REVIEW

**Comment by verythorough** [comment]
Deploy preview for *netlify-cms-www* ready!

Built with commit 61c3323d96fc83c6fd9d5e4c03e36f2c46ab2d18

https://deploy-preview-1563--netlify-cms-www.netlify.com

**Comment by verythorough** [comment]
Deploy preview for *cms-demo* ready!

Built with commit 61c3323d96fc83c6fd9d5e4c03e36f2c46ab2d18

https://deploy-preview-1563--cms-demo.netlify.com

File: packages/netlify-cms-backend-github/src/API.js [hunk 1]
```diff
@@ -557,7 +557,7 @@ export default class API {
     const contentKey = slug;
     const branchName = this.generateBranchName(contentKey);
     return this.retrieveMetadata(contentKey)
-    .then(metadata => this.closePR(metadata.pr, metadata.objects))
```
Comment (erquhart) [inline packages/netlify-cms-backend-github/src/API.js]: Looks like we _should_ be removing associated metadata when a PR is closed, but we're not.

Comment (tech4him1) [reply packages/netlify-cms-backend-github/src/API.js]: @erquhart Feel free to push a fix for this here, or separately.

**Pull request merged**